tests/ci/check_so_interface.py

```diff
@@ -10,42 +10,210 @@
         }
     },
 }
-def _compare_args(ast_args, spec_args) -> bool: #are ast_args and spec_args both dictionaries? are these in the right order? I treated ast_args like a dictionary for this
-    # check the arguments in the code match the specification
-    if len(spec_args) != len(ast_args):
-        return False
-    for expected, actual in zip(spec_args, ast_args): #expected, actual == spec, ast (a little more understandable)
-        # Check argument name
-        if expected['name'] != actual.arg:
+
+spec_args = {
+    "setup": [
+        
+    ], 
+    "set_amplifier_bias": [
+        
+    ], 
+    "set_cryo_card_ps_en": [
+        
+    ], 
+    "which_on": [
+        
+    ], 
+    "band_off": [
+
+    ], 
+    "channel_off": [
+
+    ], 
+    "full_band_resp": [
+
+    ], 
+    "find_freq": [
+
+    ],
+    "setup_notches": [
+
+    ], 
+    "run_serial_gradient_descent": [
+
+    ], 
+    "run_serial_eta_scan": [
+
+    ], 
+    "plot_tune_summary": [
+
+    ], 
+    "tracking_setup": [
+
+    ], 
+    "set_amplitude_scale_array": [
+
+    ],
+    "set_tes_bias_bipolar_array": [
+
+    ], 
+    "set_tes_bias_high_current": [
+
+    ], 
+    "set_tes_bias_low_current": [
+
+    ], 
+    "set_mode_dc" "set_mode_ac": [
+
+    ], 
+    "flux_ramp_setup": [
+
+    ],
+    "set_stream_enable": [
+
+    ], 
+    "take_stream_data": [
+
+    ], 
+    "take_noise_psd": [
+
+    ], 
+    "stream_data_on": [
+
+    ], 
+    "stream_data_off": [
+
+    ], 
+    "read_stream_data": [
+
+    ],
+    "set_downsample_filter": [
+
+    ], 
+    "run_iv": [
+
+    ], 
+    "analyze_iv": [
+
+    ]
+}
+freeze = {}
+
+with open('tests/ci/frozen_functions.py', 'r') as fh:   #opens file "fname", in read mode 'r', to be used in code as "fh"
```

Comment:
what is going to be stored in this file? Is it a pseudo-python representation of the objects/methods structure that define the interface?